src/components/UserProfile/UserProfile.tsx

```diff
@@ -8,35 +8,62 @@ import CardContent from '@mui/material/CardContent';
 import Divider from '@mui/material/Divider';
 import Typography from '@mui/material/Typography';
 import { UserProfileProps } from '../../features/profile/Profile';
+import { useAppDispatch } from '../../app/hooks';
+import { updateUserProfileAvatar } from '../../redux/profile/operations';
+import { toast } from 'react-toastify';
 
-export const UserProfile: FC<UserProfileProps> = ({ profile }) => (
-  <Card>
-    <CardContent>
-      <Box
-        sx={{
-          alignItems: 'center',
-          display: 'flex',
-          flexDirection: 'column',
-        }}
-      >
-        <Avatar
-          src={`${profile.avatar}`}
+export const UserProfile: FC<UserProfileProps> = ({ profile }) => {
+  const dispatch = useAppDispatch();
+  const [file, setFile] = useState<File>();
+
+  const handleFileChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      setFile(e.target.files[0]);
+    }
+  };
+
+  const handleUploadClick = async () => {
+    if (!file) return;
+
+    const result = await dispatch(updateUserProfileAvatar(file));
+
+    if (result.meta.requestStatus === 'fulfilled') {
+      toast.success('Avatar successfully uploaded');
+    } else {
+      toast.error('Avatar upload was failed');
+    }
+  };
+
+  return (
+    <Card>
+      <CardContent>
+        <Box
           sx={{
-            height: 80,
-            mb: 2,
-            width: 80,
+            alignItems: 'center',
+            display: 'flex',
+            flexDirection: 'column',
           }}
-        />
-        <Typography gutterBottom variant="h5">
-          {`${profile.firstName ?? ''} ${profile.lastName ?? ''}`}
-        </Typography>
-      </Box>
-    </CardContent>
-    <Divider />
-    <CardActions>
-      <Button fullWidth variant="text">
-        Upload picture
-      </Button>
-    </CardActions>
-  </Card>
-);
+        >
+          <Avatar
+            src={profile.avatar?.url ?? ''}
+            sx={{
+              height: 80,
+              mb: 2,
+              width: 80,
+            }}
+          />
+          <Typography gutterBottom variant="h5">
+            {`${profile.firstName ?? ''} ${profile.lastName ?? ''}`}
```

Comment:
in case where you are trying to do string with some set of nullable values is better to do in this way
`[profile.firstName, profile.lastName].filter(Boolean).join(' ')`
So you won't have cases like '<firstName><space>'